### ISSUE TITLE
Override CA for smarthost TLS verification

### ISSUE THREAD

**Issue by yunzheng** [OpenSMTPD/OpenSMTPD#383]
Before opensmtpd 5.4.1_p1 you could import certificates into your openssl CA store and opensmtpd would pick it up.

This is great if you want to verify your smarthosts using "relay via tls://smarthost verify" and you have your own custom self-signed CA.

Since 5.4.1_p1 this is now partly broken, you can now specify custom CA by host but it will also look for a client certificate which I do not need if I just want to verify smarthosts.

I expected the following behavior:
 * OpenSSL ca store is still the default, but it is now broken and looks for /etc/ssl/cert.pem instead (might vary by packaging)
 * Specifying a custom CA globally, without assigning it to any hosts or requiring client certificates. This would be great if you only trust your own CA.
 * Specifying a custom CA for a host should also be possible without specifying a client certificate, for example just for verifying your smarthost.

For now I have created the /etc/ssl/cert.pem containing my custom CA certificate. This works but it would be nice to override this file in my smtpd.conf. And a good fallback (or by default) would be to use the builtin OpenSSL certificate store.

I suggest a new pki directive to override the CA_FILE:

pki "*" ca "/etc/ssl/certs/myca.pem"

Or something in the likes of that.

**Comment by sysfu** [comment]
What's the status on this, has in been included in the latest release?

**Comment by poolpOrg** [comment]
I've looked at this but I'm confused:

    OpenSSL ca store is still the default, but it is now broken and looks for /etc/ssl/cert.pem instead (might vary by packaging)

Indeed, the packager decides default ca store to use with --with-ca-file

    Specifying a custom CA globally, without assigning it to any hosts or requiring client certificates. This would be great if you only trust your own CA.

Ok, just trying to understand, this is a feature request right ?
You would like to be able to setup a global CA, I get this, but does setting it redundantly with a per-host directive works ?

    Specifying a custom CA for a host should also be possible without specifying a client certificate, for example just for verifying your smarthost.

Agreed, I'll have a look at it.

**Comment by poolpOrg** [comment]
I have received a mail from another user who had a slightly different issue.

After a bit of thinking, I realized that both your issues are related, so I've opened a ticket #562 to track them both in the same place, it also contains an explanation of the problem and direction for how it's going to be fixed.

I'll close this ticket to avoid a duplicate.

The issue should be solved by this week-end, it will be part of next snapshot as well as next major release, it will need a slight configuration change.
